### PR TITLE
[cmake] Opening hours lib and tests

### PR DESCRIPTION
**Pull request by therearesomewhocallmetim**
Use 
```CC=clang CXX=clang++ cmake ../omim``` to compile on Linux

### CODE REVIEW

File: 3party/opening_hours/opening_hours_tests/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,19 @@
+set(OPENING_HOURS_INCLUDE ../opening_hours)
+
+include_directories(${PROJECT_SOURCE_DIR})
+
+
+set(
+    SRC
+    ../opening_hours.hpp
```
Comment (ygorshenin) [inline 3party/opening_hours/opening_hours_tests/CMakeLists.txt]: Why? Why do you need to depend on headers here?

File: 3party/opening_hours/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    parse_opening_hours.hpp
+    parse_opening_hours.cpp
+    rules_evaluation_private.hpp
+    rules_evaluation.hpp
+    rules_evaluation.cpp
+)
+
+add_library(opening_hours ${SRC})
+
+add_subdirectory(opening_hours_tests)
```
Comment (ygorshenin) [inline 3party/opening_hours/CMakeLists.txt]: Maybe it's worth to add this subdirectory only when NO_TEST is not set?

File: 3party/opening_hours/opening_hours_tests/CMakeLists.txt (hunk 1 above)
Comment (therearesomewhocallmetim) [inline 3party/opening_hours/opening_hours_tests/CMakeLists.txt]: In all honesty, I copied the files from the .pro file.

Comment (ygorshenin) [inline 3party/opening_hours/opening_hours_tests/CMakeLists.txt]: Bad idea. Dijsktra would not like it.

Comment (therearesomewhocallmetim) [inline 3party/opening_hours/opening_hours_tests/CMakeLists.txt]: Done

File: 3party/opening_hours/CMakeLists.txt (hunk 2 above)
Comment (therearesomewhocallmetim) [inline 3party/opening_hours/CMakeLists.txt]: Done

**Comment by therearesomewhocallmetim** [comment]
PTAL

**Comment by syershov** [comment]
LGTM

**Comment by ygorshenin** [comment]
LGTM

**Pull request merged**